### ISSUE TITLE
upload directory for the commitlog

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#2592]
If we need to move data from one node to the other, in the event of a crash or to work around a problem for example, we can use nodetool refresh.

However, some of the data for that node could still be in the commitlog. While there are ways to recover it, doing so is quite cumbersome and involved restarting the affected node out of the cluster, moving the commitlog files, restarting, forcing a flush, etc.

It just occurred to me while doing this procedure with a user, that we can have a much better way: we can have an upload directory inside the commitlog as well, and move files there directly.

nodetool refresh would then replay those files, in addition to moving the files in the data upload directory

**Comment by slivne** [comment]
Please note that commitlogs are different than sstables.

Commitlog have an RP position - which are node specific and related to the sstables on that node - so a user should not move into a commitlog directory information from another nodes commitlog info.

What we can do is from the upload directory replay the commitlog information. But we should not move the commitlog files from that directory to the parent commitlog directory. 

In addition today we support splitting the work of sharding the sstables - doing the same on commitlogs will need to be implemented.

**Comment by glommer** [comment]
You are correct - that is the direction I want to go (replaying from upload).
